.github/workflows/ci-checks.yml

```diff
@@ -0,0 +1,87 @@
+
+on:
+  push:
+    branches:
```

Comment:
Just noticed this part as I was writing my third comment - looks like it's a WIP right? I can come back to this later on then

In general looks good